slurmutils/models/model.py

```diff
@@ -114,3 +114,7 @@ def dict(self) -> Dict[str, Any]:
     def json(self) -> str:
         """Return model as json object."""
         return json.dumps(self.dict())
+
+    def override(self, other):
+        """Override the fields of this model with the fields of another model."""
+        self.data.update(other.dict())
```

Comment:
I'm wondering if `override/update` should be an abstract method instead. Reason saying is that this default implementation will not work for the `Node`, `FrontendNode`, `Partition`, etc data models.

`dict()` does not produce a dictionary that maps 1:1 with `self.data` for these models. `dict()` will inject the primary key that is used as the unique identifier, and set up `self.data` as the value for that key:

https://github.com/charmed-hpc/slurmutils/blob/6621623b3f77126c45f3cfcb7fc25835e13d15b8/slurmutils/models/slurm.py#L70-L72

Some additional logic needs to be implemented to handle the data model's primary key. The other question is do we want to even allow merging a different primary key since that effectively changes data model to a new, unique object?

Reply:
Or we could just use `data` directly?
```python
self.data.update(other.data)
```

> do we want to even allow merging a different primary key since that effectively changes data model to a new, unique object?

I'd say we do, but if we exclude mutation of the primary key. I can see it being useful to copy the settings of a `Node` into other `Nodes`, but I think we should disallow changing the name using this.